lib/customer.rb

```diff
@@ -0,0 +1,38 @@
+require "csv"
+require "awesome_print"
+
+# Build Customer class, instantiate readers for instance variables.
+class Customer
+  attr_reader :id, :all_customers, :email, :address
+
+  # Build constructor to accept parameters od id, email, and address and
+  #create instance variables.
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  # All method reads every customer from given CSV and returns a formatted array.
+  def self.all
+    all_customers = []
+    array_array_customers = CSV.read("data/customers.csv")
+    array_array_customers.each do |customer|
```

Comment:
`array_array_customers` is kind of an odd name, I would probably name it something like `customer_rows` instead.

Also, since you only reference it once you can simply inline the `CSV.read` and not bother naming it:
```suggestion
    CSV.read("data/customers.csv").each do |customer|
```